### PR TITLE
Find matching history items in notebooks

### PR DESCRIPTION
**Pull request by mtekman**
Added `find_matching_history_ids` function which can be used to find datasets in a history matching regex pattern(s).  These can then be fed into the `get` function. This should allow notebooks to be more flexible in how they import data especially with respect to workflows.

@bgruening is this function automatically exported?

### CODE REVIEW

**Comment by mtekman** [comment]
Future reference (to me), to test locally:

```
conda create -n gie pip
conda activate gie
pip install git+https://github.com/mtekman/galaxy_ie_helpers.git 
GALAXY_URL=https://usegalaxy.eu HISTORY_ID=424b38f7b103b91f API_KEY=[blahblahblah] 

~/miniconda3/envs/gie/bin/get -i ".*annotation.*" -t regex
> ['/import/3', '/import/7']
```

**Comment by mtekman** [comment]
@bgruening this looks like it works fine

File: bin/get [hunk 1]
```diff
@@ -8,9 +8,9 @@ if __name__ == '__main__':
     parser.add_argument('-i', '--id',
                         required = True,
                         nargs='+',
-                        help='The dataset ID/name from your Galaxy history.')
-    parser.add_argument('-t', '--identifier_type', dest="identifier_type", choices=['hid', 'name'], default='hid',
-                        help='Type of the argument File/ID Number. Per default, integer ID number.')
+                        help='The dataset ID/name from your Galaxy history, or a regex pattern to search all files in the history')
+    parser.add_argument('-t', '--identifier_type', dest="identifier_type", choices=['hid', 'name', 'regex'], default='hid',
```
Comment (bgruening) [inline bin/get:12]: > regex is used as the dataset ID/name

used as? used to identify the .... or similar ...

**Comment by bgruening** [comment]
Thanks.

**Pull request merged**